Sources/NetworkProtectionTestUtils/Status/MockNetworkProtectionStatusReporter.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  NetworkProtectionStatusReporter.swift
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#if os(macOS)
+
+import Combine
+import NetworkExtension
+import Common
+import NetworkProtection
+
+/// This is the default status reporter.
+///
+public final class MockNetworkProtectionStatusReporter: NetworkProtectionStatusReporter {
```

Comment:
This mock was created for share use by the macOS code which had to change as a result of this PR